rust/lance-core/src/datatypes/field.rs

```diff
@@ -547,7 +545,22 @@ impl Field {
             });
         }
 
-        Ok(self.clone())
+        Ok(if self.id >= 0 {
+            self.clone()
+        } else {
+            other.clone()
+        })
+    }
```

Comment:
These changes are a little bit squirrel-y and the whole concept of intersection ignoring data types is a little odd...I'm leaning towards special column name at this point.